packages/docusaurus-plugin-sitemap/src/index.ts

```diff
@@ -7,7 +7,7 @@
 
 import fs from 'fs-extra';
 import path from 'path';
-import {PluginOptions} from './types';
+import type {Options} from '@docusaurus/plugin-sitemap';
```

Comment:
This is not the same type on purpose 🤪 (also a good reason why one PR = one change is a good idea)

- Options is the partial type that the users are using in configuration, documented with JSDoc annotations (somehow it's TS public API surface)
- PluginOptions is the complete normalized type with defaults applied

Users should use `Options` in config

Internally, we should use `PluginOptions` because we know that normalization was already applied in docusaurus core and it's the actual type that gets injected in the plugin

I know that the naming convention is not very explicit, not sure how to find better names.

`Options` looks to be better for public API surface as it's simpler to use convention in JSDoc annotations without reading any doc.

Does it make sense?

Reply:
The changes I already made have taken that into account. These options are never normalized and are fully partial throughout.

Reply:
Oh I see

Still it might be useful to keep a convention of having 2 distinct types in each plugin so that they all look the same, even if the 2 types are the same in this case. If one day we refactor and add some option with a default value to this plugin, we'll have to revert this change, while we could just update the normalized type otherwise. Type aliases are also useful to convey meaning `type PluginOptions = Options` 🤪

Reply:
And in this case it's not 100% true either. These options are normalized, all values have defaults, and the type that the plugin constructor should receive is not partial, but a complete type with all values being set:

![image](https://user-images.githubusercontent.com/749374/146362326-90c5f169-16fa-4c38-a213-3f5229315eb0.png)

Reply:
Well—that means the types we had previously were problematic 🙈 If you look at the deleted types file, the values are still optional.

Reply:
Oh yes that might totally be the case 😄 PluginOptions should have non-optional types